srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/ProcedureCsvReader.java

```diff
@@ -0,0 +1,278 @@
+     * @param record the CSVRecord to translate
+     * @param errors to record validation errors
+     * @param rowNumber the row number for any validation errors (1-based)
+     * @return the translated Procedure, if there were no validation errors
+     */
+    private Optional<Procedure> translateRecord(
+            final CSVRecord record,
+            final Collection<TabularUploadError> errors,
+            final int rowNumber)
+    {
+        // Construct our own error list to detect if we added any errors.
+        final ArrayList<TabularUploadError> ourErrors = new ArrayList<>();
+
+        /* Get all the string values first since they don't involve exceptions */
+        
+        final String cptCode = tryGetValue(record, 0);
+        if (cptCode.length() != Procedure.CPT_CODE_LENGTH)
+        {
+            ourErrors.add(TabularUploadError.forField(
+                    rowNumber,
+                    "cptCode",
+                    String.class,
+                    ValidationCodes.BAD_FIXED_LENGTH,
+                    new Object[] { Procedure.CPT_CODE_LENGTH },
+                    "invalid length"));
+        }
+        
+        final String complexity = tryGetValue(record, 3);
+        ValidationUtils2.validateRequiredString(
+                complexity, rowNumber, "complexity", Procedure.COMPLEXITY_MAX, ourErrors);
+        
+        final String longDesc = tryGetValue(record, 4);
+        ValidationUtils2.validateRequiredString(
+                longDesc, rowNumber, "longDescription", Procedure.DESCRIPTION_MAX, ourErrors);
+        
+        final String shortDesc = tryGetValue(record, 5);
+        ValidationUtils2.validateRequiredString(
+                shortDesc, rowNumber, "shortDescription", Procedure.DESCRIPTION_MAX, ourErrors);
+        
+        final String rvuString = tryGetValue(record, 1);
+        float rvu = Float.NaN;
+        try
+        {
+            rvu = Float.parseFloat(rvuString);
+        }
+        catch (final NumberFormatException ex)
+        {
+            ourErrors.add(TabularUploadError.forField(
+                    rowNumber,
+                    "rvu",
+                    float.class,
+                    ValidationCodes.TYPE_MISMATCH,
+                    null,
+                    ex.getMessage()));
+        }
+
+        final String eligibleString = tryGetValue(record, 2);
+        boolean eligible = false;
+        try
+        {
+            eligible = parseCsvBoolean(eligibleString);
+        }
+        catch (final IllegalArgumentException ex)
+        {
+            // Error on parseCsvBoolean()
+            ourErrors.add(TabularUploadError.forField(
+                    rowNumber,
+                    "eligible",
+                    boolean.class,
+                    ValidationCodes.TYPE_MISMATCH,
+                    null,
+                    ex.getMessage()));
+        }
+            
+        // Only create the procedure if we did not record any validation ourErrors for
```

Comment:
This sentence sounds weird, like you meant to put "validation errors" instead of what is there.